simplesat/sat/minisat.py

```diff
@@ -12,6 +13,155 @@
 from .assignment_set import AssignmentSet
 from .clause import Clause
 from .policy import DefaultPolicy
+from simplesat.utils import timed_context
+
+
+class UNSAT(object):
+
+    """An unsatisfiable set of boolean clauses."""
+
+    def __init__(self, conflict, learned_clause, assignments, trails):
+        """
+        Create a new UNSAT object.
+
+        Parameters
+        ----------
+        conflict : Clause
+            The clause which has been found to be unsatisfiable.
+        learned_clause : Clause
+            The implied change needed to satisfiy the conflict.
+
+            This clause is always a learned clause and should always have
```

Comment:
We store the entire clause because it includes history information that we use to search for the requirement that caused it.